internal/token.go

```diff
@@ -931,18 +931,12 @@ loop:
 }
 
 func (z *Tokenizer) hasTag(s string) bool {
-loop:
 	for i := len(z.attr) - 1; i >= 0; i-- {
-
 		x := z.attr[i]
 		key := z.buf[x[0].Start:x[0].End]
-		for i := 0; i < len(key) && i < len(s); i++ {
-			c := key[i]
-			if c != s[i] {
-				continue loop
-			}
+		if string(key) == s {
+			return true
 		}
-		return true
 	}
 	return false
```

Comment:
This was the fix! `data-astro-raw` was previously matching any slice (from 0 to len) of `data-astro-raw`, so `data` would incorrectly be matched.

The fix was to simplify the loop logic and just do simple string comparison.

Reply:
oops

Reply:
> This was the fix! `data-astro-raw` was previously matching any slice (from 0 to len) of `data-astro-raw`, so `data` would incorrectly be matched.
> 
> The fix was to simplify the loop logic and just do simple string comparison.
I didnt expect it to be nested this deep, I didnt even knew that the golang was used for the compiler